Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -16,54 +17,81 @@ public sealed class PuddleSystem : EntitySystem
     {
         [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
         [Dependency] private readonly FluidSpreaderSystem _fluidSpreaderSystem = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        float visualSeed = 0; // given a random value on puddle initialization, to be used as a seed for rotation and sprite selection.
```

Comment:
There is no need for the system to have this variable. Just set the appearance data directly to the output of `NextFloat()`